src/routes/process-user-vocabulary/components/AudioRecordOverlay.tsx

```diff
@@ -0,0 +1,153 @@
+`
+
+const RecordingInfo = styled.Text`
+  font-size: ${props => props.theme.spacings.lg};
+  text-align: center;
+`
+
+const MeteringInfo = styled.View`
+  flex-direction: row;
+  align-self: center;
+  min-height: ${hp('12%')}px;
+`
+
+const MeteringBar = styled.View<{ height: number }>`
+  min-height: ${props => props.theme.spacings.xxs}
+  height: ${props => `${props.height * 2}`}px;
+  width: ${props => props.theme.spacings.xxs}
+  background-color: ${props => props.theme.colors.audioRecordingActive};
+  border-radius-: 50px;
+  align-self: center;
+  margin: 0 1px;
+`
+
+const InfoContainer = styled.View`
+  justify-content: flex-end;
+`
+
+interface AudioRecordOverlayProps {
+  onClose: () => void
+  onStartRecording: () => Promise<void>
+  onStopRecording: () => Promise<void>
+  recordingTime: string
+  meteringResults: number[]
+}
+
+// Zero alignment of values with the minimum metering value
+const cleanUpMeteringResults = (meteringResults: number[]): number[] => {
+  const filteredResults = meteringResults.filter(el => el !== Math.min(...meteringResults))
+  return filteredResults.map(el => el + Math.abs(Math.min(...filteredResults)))
+}
+
+const AudioRecordOverlay = ({
+  onClose,
+  onStartRecording,
+  onStopRecording,
+  recordingTime,
+  meteringResults,
+}: AudioRecordOverlayProps): ReactElement => {
+  const [isPressed, setIsPressed] = useState<boolean>(false)
+  const permissionGranted = useGrantPermissions(
+    Platform.OS === 'android'
+      ? [
+          PERMISSIONS.ANDROID.RECORD_AUDIO,
+          PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE,
+          PERMISSIONS.ANDROID.WRITE_EXTERNAL_STORAGE,
+        ]
+      : PERMISSIONS.IOS.MICROPHONE
+  )
+  const { hold, talk } = getLabels().general.audio
+  const theme = useTheme()
+
+  const cleanedMetering = useMemo(() => cleanUpMeteringResults(meteringResults), [meteringResults])
+
+  return (
+    <RNModal visible transparent animationType='fade' onRequestClose={() => onClose()}>
+      <Container>
+        <Icon onPress={() => onClose()}>
+          <CloseIcon width={theme.spacingsPlain.lg} height={theme.spacingsPlain.lg} />
+        </Icon>
+        <Content>
+          <HeadingContainer>
+            <Heading>{isPressed ? talk : hold}</Heading>
+          </HeadingContainer>
+          <InfoContainer>
+            <MeteringInfo>
+              {cleanedMetering.map((element, index) => (
+                // eslint-disable-next-line react/no-array-index-key -- no better key available
+                <MeteringBar key={index} height={element} />
+              ))}
+            </MeteringInfo>
+            <RecordingInfo>{recordingTime.slice(1, recordingTime.length)}</RecordingInfo>
+          </InfoContainer>
+          {permissionGranted && (
+            <RecordIcon
+              onPressIn={() =>
+                onStartRecording()
+                  .catch(reportError)
+                  .finally(() => setIsPressed(true))
+              }
+              onPressOut={() =>
```

Comment:
This pressout is so incredibly slow. It takes about 2 or three seconds until the recording has stopped. Maybe memoing the MeteringInfo and bar like [this](https://stackoverflow.com/questions/71844887/reactjs-memo-dynamic-for-loop-container-with-dynamic-memoized-children-how-to) could help?
Id' also prefer if the whole metering set up would happen inside the Overlay, meaning we'd need to pass the audio recorder and stuff. But feel free to argue against that. :)

Reply:
hm i tested on real devices and can not reproduce any slowness. If you pressOut the Modal is closed immediately. Tested on real device! How did you reproduce that?

Well then we would have to split the recording and play functions because playing is executed in the `UserVocabularyProcessScreen`
We also have to delete the metering results from this screen by clicking delete button.
I think its getting more complicated, even i understand that from a logical perspective it makes sense to put it in the `AudioRecordOverlay`

Reply:
i moved stuff to overlay